build-jdk.groovy

```diff
@@ -57,9 +57,7 @@ ant.propertyfile(file: 'build-jdk.properties', comment: "${name} ${version} bina
 			def url = "https://download.bell-sw.com/java/${build}/bellsoft-${type}${build}-${pkg}"
 			println url
 
-			def file = new File('cache', url.tokenize('/').last())
-			new AntBuilder().get(src: url, dest: file, skipExisting: 'yes')
-			def checksum = file.bytes.digest('SHA-256').padLeft(64, '0')
+			def checksum = new URL(url).bytes.digest('SHA-256').padLeft(64, '0')
```

Comment:
You can merge this, because Antbuilder.get throw error on my system. This simplify change works too.
   @rednoah

Reply:
This change will disable caching. For some packages we actually need to download the file and then compute the `SHA-256`. I don't know what the error says, but presumably it'll indicate that you have to create a `cache` folder first.

Reply:
Error:
```sh
make update
ant -lib lib update-jdk
Buildfile: /mnt/d/git/java-installer/build.xml

update-jdk:
   [delete] Deleting: /mnt/d/git/java-installer/build-jdk.properties
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_x64_windows_hotspot_18.0.2.1_1.zip
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_x86-32_windows_hotspot_18.0.2.1_1.zip
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_x64_mac_hotspot_18.0.2.1_1.tar.gz
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_aarch64_mac_hotspot_18.0.2.1_1.tar.gz
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_x64_linux_hotspot_18.0.2.1_1.tar.gz
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_arm_linux_hotspot_18.0.2.1_1.tar.gz
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_aarch64_linux_hotspot_18.0.2.1_1.tar.gz
[groovy:groovy] https://github.com/adoptium/temurin18-binaries/releases/download/jdk-18.0.2.1%2B1/OpenJDK18U-jdk_ppc64le_linux_hotspot_18.0.2.1_1.tar.gz
[groovy:groovy] https://download.bell-sw.com/java/18.0.2.1+1/bellsoft-jdk18.0.2.1+1-linux-i586.tar.gz
[groovy:groovy]       [get] Getting: https://download.bell-sw.com/java/18.0.2.1+1/bellsoft-jdk18.0.2.1+1-linux-i586.tar.gz
[groovy:groovy]
[groovy:groovy]       [get] To: /mnt/d/git/java-installer/cache/bellsoft-jdk18.0.2.1+1-linux-i586.tar.gz
[groovy:groovy]
[groovy:groovy]       [get] Error getting https://download.bell-sw.com/java/18.0.2.1+1/bellsoft-jdk18.0.2.1+1-linux-i586.tar.gz to /mnt/d/git/java-installer/cache/bellsoft-jdk18.0.2.1+1-linux-i586.tar.gz
[groovy:groovy]

BUILD FAILED
: java.nio.file.NoSuchFileException: /mnt/d/git/java-installer/cache/bellsoft-jdk18.0.2.1+1-linux-i586.tar.gz
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
        at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
        at java.base/java.nio.file.Files.newOutputStream(Files.java:220)
        at org.apache.tools.ant.taskdefs.Get$GetThread.downloadFile(Get.java:856)
        at org.apache.tools.ant.taskdefs.Get$GetThread.get(Get.java:694)
        at org.apache.tools.ant.taskdefs.Get$GetThread.run(Get.java:678)

Total time: 9 seconds
make: *** [makefile:4: update] Error 1
```

Little change works perfectly, i thing, that "cache" is not necesary.
```java
new URL(url).bytes.digest('SHA-256').padLeft(64, '0')
```

Reply:
Your code downloads `bellsoft-jdk18.0.2.1+1-linux-i586.tar.gz` every time, thus slow and wasteful. My code only once, thus fast. During development, you'll want to be able to test change rapidly and instantly, and not wait for a `180 MB` download, especially not if you're unlucky to be on mobile at the time.

We can add a line that will create the `cache` folder if it doesn't yet exist. I guess `ant <get>` doesn't do that automatically.

Reply:
Fixed with https://github.com/rednoah/java-installer/commit/931053478bcf4063f46809570e447f4087e293b5